src/distilabel/llm/mistralai.py

```diff
@@ -0,0 +1,157 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, List, Optional, Union
+
+from distilabel.llm import LLM
+from distilabel.llm.utils import LLMOutput
+from distilabel.logger import get_logger
+from distilabel.utils.imports import _MISTRALAI_AVAILABLE
+
+if _MISTRALAI_AVAILABLE:
+    from mistralai.client import MistralClient
+    from mistralai.models.chat_completion import ChatMessage
+
+
+if TYPE_CHECKING:
+    from distilabel.tasks.base import Task
+    from distilabel.tasks.prompt import SupportedFormats
+
+logger = get_logger()
+
+
+class MistralAILLM(LLM):
+    def __init__(
+        self,
+        task: "Task",
+        model: str = "mistral-tiny",
+        client: Optional["MistralClient"] = None,
+        api_key: Union[str, None] = None,
+        max_tokens: int = 128,
+        temperature: float = 1.0,
+        top_p: float = 1.0,
+        seed: Optional[int] = None,  # random_seed in MistralAI
+        safe_prompt: bool = False,  # Different param, https://docs.mistral.ai/platform/guardrailing/
+        num_threads: Union[int, None] = None,
+        prompt_format: Union["SupportedFormats", None] = None,
+        prompt_formatting_fn: Union[Callable[..., str], None] = None,
+    ) -> None:
+        super().__init__(
+            task=task,
+            num_threads=num_threads,
+            prompt_format=prompt_format,
+            prompt_formatting_fn=prompt_formatting_fn,
+        )
+        if not _MISTRALAI_AVAILABLE:
+            raise ImportError(
+                "`MistralAILLM` cannot be used as `mistralai` is not installed, please "
+                " install it with `pip install mistralai`."
+            )
+
+        self.max_tokens = max_tokens
+        self.temperature = temperature
+        self.top_p = top_p
+        self.seed = seed
+        self.safe_prompt = safe_prompt
+
+        # If the api key is not given, try to get it from the environment
+        if not api_key:
+            api_key = os.environ.get("MISTRALAI_API_KEY")
+        # Explicitly write the default parameters of the model
+        self.client = client or MistralClient(
+            api_key=api_key, max_retries=5, timeout=120
+        )
+        assert (
+            model in self.available_models
+        ), f"Provided `model` is not available in MistralAI, available models are {self.available_models}"
+        self.model = model
+
+    def __rich_repr__(self) -> Generator[Any, None, None]:
```

Comment:
we have a lot more params defined above

Reply:
Those are the common ones we've defined in other LLMs, no problem redefining what should we put there to make it common across the LLMs, but maybe it's better to tackle that once we refactor the LLM parameters